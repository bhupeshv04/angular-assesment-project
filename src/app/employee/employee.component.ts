import { Component, OnInit } from '@angular/core';
import { IEmployee } from '.././Interface/employee';
import { EmployeeService } from '.././Services/employee.service';
import { ActivatedRoute, Router } from '@angular/router';

@Component({
  selector: 'app-employee',
  templateUrl: './employee.component.html',
  styleUrls: ['./employee.component.css']
})
export class EmployeeComponent implements OnInit {
 employee: IEmployee;
 statusMessage: string = 'Loading data. Please wait...';
  constructor(private _employeeService: EmployeeService,
        private _activatedRoute: ActivatedRoute,
        private _router: Router) { }

  ngOnInit() {
     let empCode: string = this._activatedRoute.snapshot.params['code'];
        this.employee=this._employeeService.getEmployeeByCode(empCode);        
    }

    onBackButtonClick() :void {
    this._router.navigate(['/employees']);
}
}