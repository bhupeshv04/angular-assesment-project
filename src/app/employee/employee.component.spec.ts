import { TestBed, async } from '@angular/core/testing';
import { EmployeeComponent } from './employee.component';
import { EmployeeService } from '.././Services/employee.service';
import { ActivatedRoute, Router } from '@angular/router';

describe('EmployeeComponentTest', () => {

  let fixture;
  let employeeService;
  let employeeInstance;

    beforeEach(async(() => {
      TestBed.configureTestingModule({
        declarations: [
            EmployeeComponent
        ],
        providers:[EmployeeService, {
          provide: ActivatedRoute, useValue: {
            snapshot:{
              params: 
              { 
                code: 'emp101' 
              }
            }
           
          }
        },
        {
          provide: Router, useValue: {
            navigate: ['/employees']
          }
        }
      ]
      }).compileComponents();

      fixture=TestBed.createComponent(EmployeeComponent);
      employeeInstance = fixture.componentInstance;
      employeeService=new EmployeeService();
    }));

    test('Should display a employee', () => {
      expect(employeeInstance._activatedRoute.snapshot.params.code).toMatchSnapshot();
    });

    test('Should display a employee of code emp101', async(() => {
      let emp={
        code: 'emp101', name: 'Tom', gender: 'Male',
        annualSalary: 5500, dateOfBirth: '6/25/1988'
    }
      expect(emp.code).toMatchSnapshot();
    }));
  });