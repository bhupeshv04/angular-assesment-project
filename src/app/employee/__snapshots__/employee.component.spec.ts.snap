// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeComponentTest Should display a employee 1`] = `"emp101"`;

exports[`EmployeeComponentTest Should display a employee of code emp101 1`] = `"emp101"`;
